### PR TITLE
Fix: move general dependencies to peer-dependencies

### PR DESCRIPTION
**Pull request by eszthoff**
Fix: move react, react-dom and prop-types to be peer-dependencies and not to bundle with package.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,10 +40,12 @@
...
     "dependencies": {
         "@babel/polyfill": "7.0.0",
-        "css-vars-ponyfill": "1.16.4",
-        "prop-types": "15.6.2",
-        "react": "16.8.6",
-        "react-dom": "16.8.6"
+        "css-vars-ponyfill": "1.16.4"
+    },
+    "peerDependencies": {
+        "prop-types": ">=15.6.2",
```
Comment (carlobernardini) [inline package.json]: I'm not sure this is right, this would mean OneUI is also compatible with new major versions (with breaking changes) of these packages?

Shouldn't it be like `^15.6.2` instead? (Same for dependencies below)

**Comment by carlobernardini** [comment]
It's probably preferable to have a separate PR for the changes to the pagination component otherwise they'll be part of the same commit when the PR is squashed (actually they already are)

File: package.json [hunk 2]
```diff
@@ -41,9 +41,11 @@
     "dependencies": {
         "@babel/polyfill": "7.0.0",
         "css-vars-ponyfill": "1.16.4",
-        "prop-types": "15.6.2",
-        "react": "16.8.6",
-        "react-dom": "16.8.6"
+        "prop-types": "^15.6.2"
```
Comment (carlobernardini) [inline package.json]: Prop-types is now listed in both in dependencies and devDependencies, is that intended?

File: etc/webpack/webpack.config.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ const webpack = require('webpack');
 const OptimizeCssAssetsPlugin = require('optimize-css-assets-webpack-plugin');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const StyleLintPlugin = require('stylelint-webpack-plugin');
+const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
```
Comment (SebastiaanPasterkamp) [inline etc/webpack/webpack.config.js]: Let's put the bundle analyzer in a different PR. It's related to the _investigation_ of omitting the react libraries from the bundle, but technically a different (development) feature and irrelevant for users. The exclusion of the react libraries _is_ relevant for users.

File: src/components/Pagination/Pagination.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import React, { useCallback } from 'react';
```
Comment (SebastiaanPasterkamp) [inline src/components/Pagination/Pagination.js]: Let's put these changes in a different PR. This is a bug-fix to make Pagination work in ngui, while putting react in devDependencies is not.
